pkg/operator/configobservation/etcd/observe_etcd.go

```diff
@@ -63,6 +63,13 @@ func ObserveStorageURLs(genericListers configobserver.Listers, recorder events.R
 				errs = append(errs, addressErr)
 				continue
 			}
+			// the installer uses dummy addresses in the subnet `192.0.2.` for host-etcd endpoints
+			// this check see if etcd-bootstrap is populated with a real ip address and uses it
+			// instead of FQDN
+			if address.Hostname == "etcd-bootstrap" && !strings.Contains(address.IP, "192.0.2") {
+				etcdURLs = append(etcdURLs, "https://"+address.IP+":2379")
```

Comment:
should we also check IP is not an empty string?

Reply:
added a check for IP validity, should cover empty string!